File: module_zhihu/src/main/java/com/gas/zhihu/ui/main/mvp/MainModel.java
package com.gas.zhihu.ui.main.mvp;

import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.OnLifecycleEvent;

import com.base.lib.di.scope.ActivityScope;
import com.base.lib.integration.repository.IRepositoryManager;
import com.base.lib.mvp.BaseModel;
import com.gas.zhihu.http.ZhihuService;
import com.gas.zhihu.bean.DailyListBean;
import com.gas.zhihu.utils.MapBeanDbUtils;

import javax.inject.Inject;

import io.reactivex.Observable;
import timber.log.Timber;

@ActivityScope
public class MainModel extends BaseModel implements MainContract.Model {


    @Inject
    public MainModel(IRepositoryManager repositoryManager) {
        super(repositoryManager);
    }

    @Override
    public Observable<DailyListBean> getDailyList() {
        return mRepositoryManager.obtainRetrofitService(ZhihuService.class)
                .getDailyList();
    }

    @Override
    public int getMapDataCount(){
        return MapBeanDbUtils.getMapDataCount();
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    void onPause() {
        Timber.d("Release Resource");
    }


}

File: app/src/main/java/com/gas/app/ui/activity/main/mvp/MainModel.kt
package com.gas.app.ui.activity.main.mvp

import com.base.lib.integration.repository.IRepositoryManager
import com.base.lib.mvp.BaseModel

import com.base.lib.di.scope.ActivityScope;
import javax.inject.Inject

import com.gas.app.ui.activity.main.mvp.MainContract


/**
 * ================================================
 * Description:
 * <p>
 * Created by GasMvpTemplate on 07/22/2020 20:05
 * ================================================
 */
@ActivityScope
class MainModel @Inject constructor(repositoryManager: IRepositoryManager) : BaseModel(repositoryManager), MainContract.Model {

    override fun onDestroy() {
        super.onDestroy();
    }
}
